cumulusci/tasks/salesforce/UpdateAdminProfile.py

```diff
@@ -175,6 +176,19 @@ def _set_tabs_visibility(self):
             elem.find("sf:visibility", self.namespaces).text = "DefaultOn"
 
     def _deploy_metadata(self):
-        self.logger.info("Deploying updated Admin.profile from {}".format(self.tempdir))
-        api = self._get_api(path=self.tempdir)
+        deploy_dir = tempfile.mkdtemp()
```

Comment:
Don't we need to clean up `deploy_dir` after deployment?

Reply:
Actually we already are working in self.tempdir; it doesn't look to me like there's a need to create another temporary directory.

We could handle the self.tempdir cleanup better in _run_task though, by using tempfile.TemporaryDirectory as a context manager.

Reply:
Looks like the consensus is that my general approach makes sense. I can take a second pass at making it a bit smarter about how we're handling the temporary directory and re-request review.